src/Concordium/Client/Types/TransactionStatus.hs

```diff
@@ -25,6 +27,31 @@ data TransactionStatusResult' a = TransactionStatusResult
 
 type TransactionStatusResult = TransactionStatusResult' ValidResult
 
+-- |Convert an `TransactionStatus` instance into a `TransactionStatusResult` instance.
+-- Returns @Left@ wrapping an error message if a transaction summary was @Nothing@ if
+-- the input is either @Committed@ or @Finalized@, or a @Right@ wrapping the result
+-- otherwise.
```

Comment:
Something seems off in this sentence. Please revise it. Perhaps it just missing an `or` after `@Nothing@`, but it could also be made a bit clearer.

Reply:
Yea, I should revise the comments. It seems there are some slip-ups here and there. Will do.

Reply:
I added a TODO to the issue.